src/test/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionServiceTest.java

```diff
@@ -95,37 +102,78 @@
     @Mock
     private OracleQueryClient oracleQueryClient;
 
+    @Mock
+    private ActionCodeValidator actionCodeValidator;
+
     @InjectMocks
     private ObjectionService objectionService;
 
     @Test
-    void createObjectionTest() {
+    void createObjectionTest() throws ValidationException {
         Objection returnedEntity = new Objection.Builder()
                 .withCompanyNumber(COMPANY_NUMBER)
                 .build();
         returnedEntity.setId(OBJECTION_ID);
         returnedEntity.setCreatedOn(MOCKED_TIME_STAMP);
-        returnedEntity.setStatus(ObjectionStatus.OPEN);
+        returnedEntity.setStatus(OPEN);
         CreatedBy createdBy = new CreatedBy(AUTH_ID, E_MAIL);
         returnedEntity.setCreatedBy(createdBy);
 
         when(objectionRepository.save(any())).thenReturn(returnedEntity);
         when(localDateTimeSupplier.get()).thenReturn(MOCKED_TIME_STAMP);
         when(ericHeaderParser.getEmailAddress(AUTH_USER)).thenReturn(E_MAIL);
+        when(oracleQueryClient.getCompanyActionCode(COMPANY_NUMBER)).thenReturn(ACTION_CODE_OK);
 
         Objection objectionResponse = objectionService.createObjection(REQUEST_ID, COMPANY_NUMBER, AUTH_ID, AUTH_USER);
 
         verify(objectionRepository).save(any());
         verify(oracleQueryClient).getCompanyActionCode(COMPANY_NUMBER);
-        
+        verify(actionCodeValidator).validate(ACTION_CODE_OK, REQUEST_ID);
+
+        ArgumentCaptor<Objection> saveObjectionCaptor = ArgumentCaptor.forClass(Objection.class);
+        verify(objectionRepository, times(1)).save(saveObjectionCaptor.capture());
+
+        Objection savedObjection = saveObjectionCaptor.getValue();
+        assertEquals(COMPANY_NUMBER, savedObjection.getCompanyNumber());
+        assertEquals(MOCKED_TIME_STAMP, savedObjection.getCreatedOn());
+        assertEquals(AUTH_ID, savedObjection.getCreatedBy().getId());
+        assertEquals(E_MAIL, savedObjection.getCreatedBy().getEmail());
+        assertEquals(REQUEST_ID, savedObjection.getHttpRequestId());
+        assertEquals(ACTION_CODE_OK, savedObjection.getActionCode());
+        assertEquals(OPEN, savedObjection.getStatus());
+
         assertEquals(OBJECTION_ID, objectionResponse.getId());
         assertEquals(MOCKED_TIME_STAMP, objectionResponse.getCreatedOn());
         assertEquals(COMPANY_NUMBER, objectionResponse.getCompanyNumber());
-        assertEquals(ObjectionStatus.OPEN, objectionResponse.getStatus());
+        assertEquals(OPEN, objectionResponse.getStatus());
         assertEquals(AUTH_ID, objectionResponse.getCreatedBy().getId());
         assertEquals(E_MAIL, objectionResponse.getCreatedBy().getEmail());
     }
 
+    @Test
+    void createObjectionIneligibleStatusTest() throws ValidationException {
```

Comment:
Is there a test scenario for 209 ticket where the code is neither eligible nor ineligible - is that covered anywhere on another ticket perhaps?

Reply:
The AllowedValuesValidationRuleTest should cover that scenario, which basically checks that if a value is in the list of 'ok' values, it is ok, if it is not in the list then a validation exception gets thrown.